File: examples/modulith-clinical-tool-jpa/src/main/java/io/zenwave360/example/clinicaltool/modules/clinical/core/outbound/jpa/ProvisionalPatientRepository.java
package io.zenwave360.example.clinicaltool.modules.clinical.core.outbound.jpa;

import io.zenwave360.example.clinicaltool.modules.clinical.core.domain.*;
import java.math.*;
import java.time.*;
import java.util.*;
import org.springframework.data.jpa.repository.*;
import org.springframework.stereotype.Repository;

/**
 * Spring Data JPA repository for the ProvisionalPatient entity.
 */
@SuppressWarnings("unused")
@Repository
public interface ProvisionalPatientRepository extends JpaRepository<ProvisionalPatient, Long> {
    java.util.Optional<ProvisionalPatient> findByPhoneNumberAndHisNumber(String phoneNumber, String hisNumber);
}

File: examples/modulith-klinical-tool-jpa/src/main/kotlin/io/zenwave360/example/clinicaltool/modules/clinical/core/outbound/jpa/ProvisionalPatientRepository.kt
package io.zenwave360.example.clinicaltool.modules.clinical.core.outbound.jpa

import io.zenwave360.example.clinicaltool.modules.clinical.core.domain.*
import java.math.*
import java.time.*
import java.util.*

import org.springframework.data.jpa.repository.*
import org.springframework.stereotype.Repository

/**
 * Spring Data JPA repository for the ProvisionalPatient entity.
 */
@Suppress("unused")
@Repository
interface ProvisionalPatientRepository : JpaRepository<ProvisionalPatient, Long> {fun findByPhoneNumberAndHisNumber(phoneNumber: String, hisNumber: String): java.util.Optional<ProvisionalPatient>
}
